src/py_zfs_pool.c

```diff
@@ -28,378 +24,252 @@ int py_zfs_pool_init(PyObject *type, PyObject *args, PyObject *kwds) {
-	Py_RETURN_NONE;
-}
+PyObject *py_zfs_pool_upgrade(PyObject *self, PyObject *args) {
+	int ret = 0;
+	py_zfs_pool_t *p = (py_zfs_pool_t *)self;
+	py_zfs_error_t err;
+
+	Py_BEGIN_ALLOW_THREADS
+	PY_ZFS_LOCK(p->pylibzfsp);
+	ret = zpool_upgrade(p->zhp, SPA_VERSION);
+	if (ret)
+		py_get_zfs_error(p->pylibzfsp->lzh, &err);
+	PY_ZFS_UNLOCK(p->pylibzfsp);
+	Py_END_ALLOW_THREADS
+
+	if (ret) {
+		set_exc_from_libzfs(&err, "zpool_upgrade() failed");
+		return (NULL);
+	}
 
-static
-PyObject *py_zfs_pool_get_guid(py_zfs_prop_t *self, void *extra) {
 	Py_RETURN_NONE;
 }
 
 static
-PyObject *py_zfs_pool_get_hostname(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
+PyObject *py_zfs_pool_delete(PyObject *self, PyObject *args, PyObject *kwargs) {
+	int ret, force;
+	py_zfs_pool_t *p = (py_zfs_pool_t *)self;
+	py_zfs_error_t err;
+	ret = force = 0;
+	char *kwnames[] = {"force", NULL};
 
-PyObject *py_zfs_pool_get_state(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "|p", kwnames, &force)) {
+		return NULL;
+	}
 
-static
-PyObject *py_zfs_pool_get_status_code(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
+	Py_BEGIN_ALLOW_THREADS
+	PY_ZFS_LOCK(p->pylibzfsp);
+	ret = zpool_disable_datasets(p->zhp, force);
+	if (ret == 0)
+		ret = zpool_destroy(p->zhp, "destroy");
+	if (ret)
+		py_get_zfs_error(p->pylibzfsp->lzh, &err);
+	PY_ZFS_UNLOCK(p->pylibzfsp);
+	Py_END_ALLOW_THREADS
+
+	if (ret) {
+		set_exc_from_libzfs(&err, "zpool_delete() failed");
+		return (NULL);
+	}
 
-static
-PyObject *py_zfs_pool_get_healthy(py_zfs_prop_t *self, void *extra) {
 	Py_RETURN_NONE;
 }
 
 static
-PyObject *py_zfs_pool_get_warning(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
+PyObject *py_zfs_pool_ddt_prefetch(PyObject *self, PyObject *args) {
+	int ret = 0;
+	py_zfs_pool_t *p = (py_zfs_pool_t *)self;
+	py_zfs_error_t err;
+
+	Py_BEGIN_ALLOW_THREADS
+	PY_ZFS_LOCK(p->pylibzfsp);
+	ret = zpool_prefetch(p->zhp, ZPOOL_PREFETCH_DDT);
+	if (ret)
+		py_get_zfs_error(p->pylibzfsp->lzh, &err);
+	PY_ZFS_UNLOCK(p->pylibzfsp);
+	Py_END_ALLOW_THREADS
+
+	if (ret) {
+		set_exc_from_libzfs(&err, "zpool_ddt_prefetch() failed");
+		return (NULL);
+	}
 
-static
-PyObject *py_zfs_pool_get_status_detail(py_zfs_prop_t *self, void *extra) {
 	Py_RETURN_NONE;
 }
 
 static
-PyObject *py_zfs_pool_get_error_count(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
+PyObject *py_zfs_pool_ddt_prune(PyObject *self,
+				PyObject *args,
+				PyObject *kwargs) {
+	int ret, days, percentage;
+	uint64_t value = 0;
+	py_zfs_pool_t *p = (py_zfs_pool_t *)self;
+	py_zfs_error_t err;
+	zpool_ddt_prune_unit_t unit = ZPOOL_DDT_PRUNE_NONE;
+	char *kwnames[] = {"days", "percentage", NULL};
+	ret = days = percentage = 0;
 
-static
-PyObject *py_zfs_pool_get_properties(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "|ii", kwnames, &days,
```

Comment:
Any reason to not use `I` (uint)? It looks like we don't allow negative values per validation below and it could simplify checks slightly.

Reply:
Updated to use `I` and simplified the check.